src/node.rs

```diff
@@ -244,12 +244,20 @@ impl<S: StorageExt> SawtoothRaftNode<S> {
 
         let is_leader = self.raw_node.raft.state == raft::StateRole::Leader;
         if is_leader {
-            // We just became the leader, so we need to start building a block
+            // We just became the leader
             if self.leader_state.is_none() {
-                debug!("Leader({:?}) became leader, intializing block", self.peer_id);
+                // Check if we stepped up as leader from follower state
+                if let Some(FollowerState::Committing(ref block_id)) = self.follower_state {
```

Comment:
Does the other `FollowerState` values require any special actions? Or is it safe to just start a new block in those cases?

Reply:
Follower states is either Idle or Committing or None. No other cases I came across yet which needs special handling other than when follower transitions to leader when it is in Committing state.

Reply:
The other FollowerState is Idle. There's a case where the node received raft log entry, added it to the log, but is not committed. Log entry is not committed because leader which proposed the block has not yet confirmed about its commit status.
The new leader election can safely change leader status in this case.  Because raft will take care if the log entry needs consideration.